tests/p2p/test_discovery.py

```diff
@@ -431,6 +450,14 @@ def get_discovery_protocol(seed=b"seed", address=None):
     return discovery.DiscoveryProtocol(privkey, address, [], CancelToken("discovery-test"))
 
 
+async def get_listening_discovery_protocol(event_loop):
+    addr = kademlia.Address('127.0.0.1', random.randint(1024, 9999))
```

Comment:
Just noting that I looked (briefly) and it appears that there are no obvious ways to get a random open UDP port